tuplex/codegen/include/LLVMEnvironment.h

```diff
@@ -76,6 +76,12 @@ namespace tuplex {
             std::unique_ptr<llvm::Module> _module;
             std::map<python::Type, llvm::Type *> _generatedTupleTypes;
             std::map<python::Type, llvm::Type *> _generatedListTypes;
+            std::map<python::Type, llvm::Type *> _generatedIterIteratorTypes;
+            // use llvm struct member types for map key since argType may contain iterator types, and iterators with the same yieldType may have different llvm structs
+            std::map<std::vector<llvm::Type *>, llvm::Type *> _generatedZipIteratorTypes;
```

Comment:
do we need to distinguish between zip / enumerate iterator types or is there some scenario where we have just one array of iterator types?

Reply:
No, here we don't need to since LLVM struct memberTypes is used as map key.
Changed to use a single map for storing both types.